src/pkgcheck/checks/python.py

```diff
@@ -240,6 +252,20 @@ def check_pep517_depend(self, item):
 
         return True
 
+    def check_pep517_mode(self, item):
```

Comment:
What do you think about merging the code here with `check_pep517_depend` and rename the function to `check_pep517`?

Then instead of using `return {True,False}`, use `yield` of corresponding result and when calling the function (after if of inherit class) do `yield from self.check_pep517(...)`?

We have very similar code between the two, and is somewhat sad we traverse the tree-sitter query 2 times.